src/goto-symex/goto_symex_state.h

```diff
@@ -185,6 +185,18 @@ class goto_symex_statet
      *  resulting function invocations with. */
     expr2tc orig_func_ptr_call;
 
+    struct
+    {
+      std::vector<std::pair<std::string, unsigned>> variables;
+      unsigned total = 0;
+
+      void add_decl(code_decl2tc s)
+      {
+        total += s->type->get_width();
+        variables.emplace_back(s->value.as_string(), s->type->get_width());
+      }
+    } frame_info;
+
```

Comment:
You can use statet::framet instead of creating a new struct:
https://github.com/esbmc/esbmc/blob/master/src/goto-symex/goto_symex_state.h#L148

Also, do you really need the keep track of the variable name?

Reply:
This is a inner struct inside framet.

About the variable name... not now. But I am guessing that in the future it will be useful for counterexample purposes.

Reply:
But you still don't need the vector of variables or the struct:
1. `total` can be an attribute of framet (an it needs a better name).
2. Since you always call `add_decl` when creating the claim, you can simply add the name to the claim:
```
void goto_symext::process_stack_size(expr2tc &expr)
{
  const code_decl2t &decl_code = to_code_decl2t(expr);
  cur_state->top().frame_info.add_decl(decl_code); /* I. REMOVE THIS */
  if(stack_limit != 0)
  {
    unsignedbv_type2tc u32type(32);
    BigInt f_size(cur_state->top().frame_info.total);
    BigInt s_size(stack_limit);
    constant_int2tc function_irep2(u32type, f_size);
    constant_int2tc limit_irep2(u32type, s_size);

    lessthanequal2tc check(function_irep2, limit_irep2);

    /* II. ADD THE NAME OF THE VARIABLE TO THE CLAIM */
    claim(check, "Stack limit property was violated when declarin " + to_decl_code(expr).value.as_string());
  }
}
```

I've added the two comments in uppercase so they are easier to spot.

Reply:
The second comment is really nice. 

Now we get the variable name in the claim:

```
Counterexample:

State 1 file main.c line 8 function main thread 0
----------------------------------------------------
Violated property:
  file main.c line 8 function main
  Stack limit property was violated when declaring xPtr

VERIFICATION FAILED
```
For this program:
```
  1 int main() {
  2   int y = 5;  // 4 * 8 = 32
  3   int *yPtr;  // 32
  4   // Expected total here: 32+32= 64.
  5 
  6   yPtr = &y;
  7 
  8   int *xPtr; // 32
  9   // The total now is 64 + 32 = 96.
 10   return 0;
 11 }
```

Now, in `symex_decl`, we have the follwoing:

```
  if(stack_limit != 0)
  {
    const std::string pretty_name =
      to_code_decl2t(code).value.as_string().substr(
        to_code_decl2t(code).value.as_string().find_last_of('@') + 1);

    claim(
      (cur_state->top().process_stack_size(code2, stack_limit)),
      "Stack limit property was violated when declaring " + pretty_name);
  }
```

Reply:
I agree with @mikhailramalho 

I have removed the struct and variables vector from framet.

I have also changed the variable `total` to `stack_frame_total`.